File: src/main/java/com/goudourasv/courses/controller/dto/CourseCreate.java
package com.goudourasv.courses.controller.dto;

import com.fasterxml.jackson.annotation.JsonProperty;

import javax.validation.constraints.NotBlank;


public class CourseCreate {
    @NotBlank
    private String title;
    @JsonProperty("institution")
    @NotBlank
    private String institutionName;
    private String tag;
    @NotBlank
    private String instructor;

    public CourseCreate(String title,String institutionName,String tag,String instructor){
        this.title = title;
        this.institutionName = institutionName;
        this.tag = tag;
        this.instructor = instructor;
    }


    public String getTitle() {
        return title;
    }

    public String getInstitutionName() {
        return institutionName;
    }

    public String getTag() {
        return tag;
    }

    public String getInstructor() {
        return instructor;
    }


}

File: src/main/java/com/goudourasv/http/courses/dto/CourseCreate.kt
package com.goudourasv.http.courses.dto

import com.fasterxml.jackson.annotation.JsonProperty
import com.goudourasv.domain.lectures.Lecture
import java.time.Instant
import java.util.*
import javax.validation.constraints.NotBlank

data class CourseCreate(
    @NotBlank val title: String,
    val startDate: Instant,
    val endDate: Instant,
    @JsonProperty("institution")
    val institutionId: UUID,
    @JsonProperty("instructor")
    val instructorId: UUID,
    val tags: MutableSet<String>? = mutableSetOf(),
    @JsonProperty("lectures")
    val lectures: List<Lecture>?

)
